recipes/ffmpeg/all/conanfile.py

```diff
@@ -176,6 +188,66 @@ def validate(self):
         if self.options.with_ssl == "securetransport" and not tools.is_apple_os(self.settings.os):
             raise ConanInvalidConfiguration("securetransport is only available on Apple")
 
+        if not self.options.avcodec:
```

Comment:
maybe it's easier to have a map of dependencies in some form, like:
```
{"avcodec": ["avdevice", "avformat"],
  "avformat": [avdevice", "avfilter"],
  ...
```
and then iterate it programmatically? might be less repeated code

Reply:
Yes, that would make sense indeed. I'll try to put something together.